inc/class-purge-cache.php

```diff
@@ -0,0 +1,183 @@
+<?php
+/**
+ * WP Irving Cache.
+ *
+ * @package WP_Irving
+ */
+
+namespace WP_Irving;
+
+/**
+ * Class to purge cache.
+ */
+class Purge_Cache {
+
+	/**
+	 * Class constructor.
+	 */
+	public function __construct() {
+		// Register admin page.
+		add_action( 'admin_menu', [ $this, 'register_admin' ] );
+
+		// Purging actions.
+		add_action( 'wp_insert_post', [ $this, 'on_update_post' ], 10, 2 );
+		add_action( 'transition_post_status', [ $this, 'on_post_status_transition' ], 10, 3 );
+		add_action( 'before_delete_post', [ $this, 'on_before_delete_post' ] );
+
+		add_action( 'init', [ $this, 'purge_cache_request' ] );
+	}
+
+	/**
+	 * Clear cache on post update.
+	 *
+	 * @param int     $post_id Post ID.
+	 * @param WP_Post $post    Post object.
+	 */
+	public function on_update_post( $post_id, $post ) {
+		if ( 'publish' !== $post->post_status ) {
+			return;
+		}
+
+		// Purge cache.
+		$this->fire_purge_request( $post_id );
+	}
+
+	/**
+	 * Clear cache on page transition.
+	 *
+	 * @param string  $new_status New Status.
+	 * @param string  $old_status Old Status.
+	 * @param WP_Post $post       Post object.
+	 */
+	public function on_post_status_transition( $new_status, $old_status, $post ) {
+		if ( 'publish' !== $new_status && 'publish' !== $old_status ) {
+			return;
+		}
+
+		// Purge cache.
+		$this->fire_purge_request( $post->ID );
+	}
+
+	/**
+	 * Clear on post delete.
+	 *
+	 * @param int $post_id Post ID.
+	 */
+	public function on_before_delete_post( $post_id ) {
+		$this->fire_purge_request( $post_id );
+	}
+
+	/**
+	 * Fire purge request.
+	 *
+	 * @param int    $post_id   Post ID.
+	 * @param string $permalink Permalink.
+	 */
+	protected function fire_purge_request( $post_id = 0, $permalink = '' ) {
+
+		// Get post permalink with fallback.
+		if ( empty( $permalink ) ) {
+			$permalink = get_the_permalink( $post_id );
+		}
```

Comment:
- If `$post_id` is `0`, this would purge the global `$post`, which might be unexpected. 
- An invalid post ID will return `false`. What would that purge?
- What if the post is now in the trash? The permalink will include the `__trashed` suffix. Do the cache implementations know how to handle those?